relay/network_graph/alg.py

```diff
@@ -79,6 +79,7 @@ def _least_cost_path_helper(
 ):
     graph_adj = graph.adj
 
+    nodes_done = set()  # set of nodes, where we already found the minimal path
```

Comment:
I would rename it visited_nodes since it is the set of nodes the algorithm has already visited